weave.lisp

```diff
@@ -22,9 +22,39 @@
 ; One-to-one correspondence between .lit and .html files.
 ; The body of an .html file should look the same regardless of whether the .lit is in another context (like a book).
 
-(defstruct weaver 
+(defun get-section (weaver section)
+  (trie-get (weaver-section-map weaver) section))
+
+(defun add-section (weaver section anchor)
+  (setf (weaver-section-map weaver)
+        (trie-put (weaver-section-map weaver) section anchor)))
+
+(comment
+ (let* ((file-defs (parse-lit-files '("dev.lit")))
+        (weaver (make-weaver-default file-defs)))
+   (add-section weaver '("Some Chapter") "c2")
+   (add-section weaver '("Some Chapter" "some-section") "s2:1")
+   (trie-get (weaver-section-map weaver) '("some-chapter"))
+   (weaver-section-map weaver))
+   ; => (("Some Chapter"
+   ;      ("some-section" (:TERMINAL . "s2:1"))
+   ;      (:TERMINAL . "c2")))
+ )
+
+(defstruct weaver
   (def-table (make-hash-table :test #'equal) :type hash-table)
   (use-table (make-hash-table :test #'equal) :type hash-table)
+  ;; section-map is used to map chapter/section text to their respective
+  ;; anchor ids, like #s2:4
+  ;;
+  ;; This lets you create markdown links like this:
+  ;; [some section or chapter](@some section or chapter)
+  ;; Which will be woven to (for example):
+  ;; [some section or chapter](#s2:4)
+  ;;
+  ;; Right now, it doesn't work across files in a book.
+  ;; Heck... it barely works at all.
+  (section-map '() :type list)
```

Comment:
Be conscious of multiple files. When a chapter/section is referenced in the same file it should be a relative link (just the anchor portion). When it is referenced in a different file, that file name will be prepended to the anchor. See how block reference anchors are produced.